### PR TITLE
aks/: Update Azure deployment files

### PR DESCRIPTION
**Pull request by nuclearcat**
Update Azure deployment files, to automate deployment.

### CODE REVIEW

File: kube/aks/README.md [hunk 1]
```diff
@@ -29,11 +29,20 @@ portal](https://portal.azure.com/#create/Microsoft.AKS).  The standard settings
...
-This AKS reference deployment relies on an Atlas account for MongoDB in order
-to keep everything in the Cloud and application setup to the bare minimal.  As
-such, please [create an Atlas
+## MongoDB/Atlas database
+
+### Self-hosted MongoDB
+
+By default script api-initial-deploy.sh install self-hosted MongoDB,
+and store data in PVC.
+Adjust size of PVC in deploy/pvc.yaml.
```
Comment (gctucker) [inline kube/aks/README.md:39]: I thought we were planning to rely on Atlas for the production deployment?  Having "self-hosted" MongoDB only applies to the vendor-agnostic Kubernetes files with e.g. MiniKube.

Comment (nuclearcat) [reply kube/aks/README.md:39]: Production yes, but for staging we can use self-hosted MongoDB (also to reduce costs), so this script planned to be used for both options.

Comment (gctucker) [reply kube/aks/README.md:39]: For staging we have docker-compose.  This seems to be mixing a lot of things up I think.

**Review by pawiecz — COMMENTED** [review]
Please let me know if I missed a PR that moves README/any docs to thew deployment repo

File: kube/aks/api.yaml [hunk 2]
```diff
@@ -7,7 +8,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: api
-  namespace: kernelci-api
+  namespace: kernelci-api-testns
```
[the remark this answers is not in the document]
Comment (nuclearcat) [reply kube/aks/api.yaml]: This namespace is updated by deploy script from kernelci-deploy. I am updating it now to match production deploy.

File: kube/aks/configmap.yaml.example [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2023 Collabora Limited
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
```
[the remark this answers is not in the document]
Comment (nuclearcat) [reply kube/aks/configmap.yaml.example:4]: Some initial recipes/configs was created by Jeny Sadadia, i reused them mostly without update

File: kube/aks/README.md [hunk 4]
```diff
@@ -1,394 +1,6 @@
...
-[https://kernelci-api.eastus.cloudapp.azure.com/latest/docs](https://kernelci-api.eastus.cloudapp.azure.com/latest/docs)
-
-## Enjoy!
-
-Now you have a publicly available API instance suitable for production.  To
-make best use of it, you can try the `kci` command line tools to do things by
-hand and run the `kernelci-pipeline` services to automatically run jobs, send
-email reports etc.  See the main documentation for pointers with more details.
+This files designed to be used by api-pipeline-deploy.sh script from [kernelci-deploy](https://github.com/kernelci/kernelci-deploy) repository.
+Additional documentation can be found in [kernelci-deploy README](https://github.com/kernelci/kernelci-deploy/kubernetes/README.md).
```
[the remark this answers is not in the document]
Comment (nuclearcat) [reply kube/aks/README.md:6]: Added back missing file, but still i will update README to latest code in follow up PR.

File: kube/aks/api.yaml [hunk 5]
```diff
@@ -19,35 +20,44 @@ spec:
...
+            - api.main:app
+            - --port
+            - '8000'
+            - --host
+            - 0.0.0.0
+          env:
+            - name: SECRET_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: kernelci-api-secret
```
[the remark this answers is not in the document]
Comment (nuclearcat) [reply kube/aks/api.yaml:40]: I verified it over making monday deploy using code from this PR. secrets are set in kernelci-deploy repository script

File: kube/aks/ingress.yaml
[the remark this answers is not in the document]
Comment (nuclearcat) [reply kube/aks/ingress.yaml:1]: I verified it over monday deploy and on staging kubernetes cluster. Quite tricky (2 ingress in different namespaces with single cert issuer) but it worked.

**Review by a-wai — APPROVED** [review]

**Pull request merged**